cluster/examples/storage/azure/account-resource-class.yaml

```diff
@@ -0,0 +1,25 @@
+# ResourceClass that defines the blueprint for how a "standard" Azure storage
+# account should be dynamically provisioned
+apiVersion: core.crossplane.io/v1alpha1
+kind: ResourceClass
+metadata:
+  name: standard-azure-account
+  namespace: crossplane-system
+parameters:
+  resourceGroupName: group-westus-1
+  storageAccountSpec: >
```

Comment:
i don't think we've used this pattern yet before where a spec is inlined into a parameter for a resource class. what drove the design that way instead of using just key value pairs?  does this make more sense to be yaml instead of json?

Reply:
As you recall we have made a design decision to represent `parameters` by `map[string]string`. 
Thus, every property key is a string, and every property value is a string. This works out nicely for flat properties structure like: `"foo":"bar"`. 

The issue gets more interesting when the `value` fields is not modeled as a single value. We already had an experience with `List` types where we encode list values as CSV string like `"foo":"bar,baz"` (Note, we were lucky so far that we didn't come across values that contain ',' )

The issue get even more interesting when value is a more complicated structure. You can find us doing some "gymnastics" with parsing values that are also 'map' structure:
https://github.com/crossplaneio/crossplane/blob/master/pkg/apis/azure/cache/v1alpha1/redis_types.go#L197
https://github.com/crossplaneio/crossplane/blob/master/pkg/apis/azure/cache/v1alpha1/redis_types.go#L222

This issue gets even ... well, you get the gist. You can only get so far by attempting to encode structured data as a semi-plain string. I am not `100%` happy by encoding some complex properties as a `json` string, but it is the simplest and ironically most straight forward implementation. 

In this specific case: 
```yaml
  storageAccountSpec: >
    {
      "kind":"Storage",
      "location":"West US",
      "sku":{
        "name":"Standard_LRS",
        "tier":"Standard"
      },
      "tags":{
        "application":"crossplane"
      }
    }
```
`storageAccountSpec` value is a `json` encoded string.

Could we encode this string as `yaml`, I think we could. I would argue it would not make more sense though. If we accept that we default to `yaml` in describing our K8S resources, I would argue, representing value as a `json` blob make is arguably less confusing, i.e. it is easy to spot that this is `json` string value inside the `yaml`. Of course, this perceived advantage goes out of the door, once we do `kubectl get -o json` :) - then you better be on the lookout for that `>` mark.